src/reducer-immutable.js

```diff
@@ -0,0 +1,40 @@
+
+import warning from 'warning';
+
+import createReducer from './utils/createReducer';
+import {
+  SHOW,
+  REPLACE,
+  DISMISS,
+} from './actions';
+
+let reducer; // eslint-disable-line import/no-mutable-exports
+
+try {
+  const fromJS = require('immutable').fromJS; // eslint-disable-line global-require
+
+  const initialState = fromJS({
+    show: false,
+    title: '',
+  });
+
+  reducer = createReducer(initialState, {
```

Comment:
I was just looking at that! I think you're right. I knew there had to be a better way.